backends/bmv2/psa_switch/midend.cpp

```diff
@@ -90,8 +90,18 @@ PsaSwitchMidEnd::PsaSwitchMidEnd(CompilerOptions& options) : MidEnd(options) {
         new P4::EliminateNewtype(&refMap, &typeMap),
         new P4::EliminateSerEnums(&refMap, &typeMap),
         new P4::RemoveActionParameters(&refMap, &typeMap),
-        convertEnums,
-        new VisitFunctor([this, convertEnums]() { enumMap = convertEnums->getEnumMapping(); }),
+    });
```

Comment:
If you load the json after the mid-end you don't need to modify the mid-end.
The fact that you do indicates that something is wrong.